src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -0,0 +1,508 @@
+        {
+            MaxCoolSetpointLimit = AbsMaxCoolSetpointLimit;
+        }
+        Requested = static_cast<int16_t>(chip::Encoding::LittleEndian::Get16(value));
+        if (Requested < AbsMinCoolSetpointLimit || Requested < MinCoolSetpointLimit || Requested > AbsMaxCoolSetpointLimit ||
+            Requested > MaxCoolSetpointLimit)
+            status = EMBER_ZCL_STATUS_INVALID_VALUE;
+        else
+            status = EMBER_ZCL_STATUS_SUCCESS;
+
+        break;
+    }
+
+    case Ids::MinHeatSetpointLimit:
+    case Ids::MaxHeatSetpointLimit: {
+        int16_t AbsMinHeatSetpointLimit; // 7C (44.5 F) is the default
+        int16_t AbsMaxHeatSetpointLimit; // 30C (86 F) is the default
+
+        status = GetAbsMinHeatSetpointLimit(endpoint, &AbsMinHeatSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMinHeatSetpointLimit = 700;
+        }
+
+        status = GetAbsMaxHeatSetpointLimit(endpoint, &AbsMaxHeatSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMaxHeatSetpointLimit = 3000;
+        }
+
+        Requested = static_cast<int16_t>(chip::Encoding::LittleEndian::Get16(value));
+        if (Requested < AbsMinHeatSetpointLimit || Requested > AbsMaxHeatSetpointLimit)
+            status = EMBER_ZCL_STATUS_INVALID_VALUE;
+        else
+            status = EMBER_ZCL_STATUS_SUCCESS;
+
+        break;
+    }
+    case Ids::MinCoolSetpointLimit:
+    case Ids::MaxCoolSetpointLimit: {
+        int16_t AbsMinCoolSetpointLimit; // 7C (44.5 F) is the default
+        int16_t AbsMaxCoolSetpointLimit; // 30C (86 F) is the default
+
+        status = GetAbsMinCoolSetpointLimit(endpoint, &AbsMinCoolSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMinCoolSetpointLimit = 700;
+        }
+
+        status = GetAbsMaxCoolSetpointLimit(endpoint, &AbsMaxCoolSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMaxCoolSetpointLimit = 3000;
+        }
+
+        Requested = static_cast<int16_t>(chip::Encoding::LittleEndian::Get16(value));
+        if (Requested < AbsMinCoolSetpointLimit || Requested > AbsMaxCoolSetpointLimit)
+            status = EMBER_ZCL_STATUS_INVALID_VALUE;
+        else
+            status = EMBER_ZCL_STATUS_SUCCESS;
+
+        break;
+    }
+
+    case Ids::ControlSequenceOfOperation: {
+        uint8_t RequestedCSO;
+        RequestedCSO = *value;
+        if (RequestedCSO > EMBER_ZCL_THERMOSTAT_CONTROL_SEQUENCE_COOLING_AND_HEATING_WITH_REHEAT)
+            status = EMBER_ZCL_STATUS_INVALID_VALUE;
+        else
+            status = EMBER_ZCL_STATUS_SUCCESS;
+
+        break;
+    }
+
+    case Ids::SystemMode: {
+        uint8_t ControlSequenceOfOperation = 0xff; // Invalid value
+        uint8_t RequestedSystemMode        = 0xff; // Invalid value
+        GetControlSequenceOfOperation(endpoint, &ControlSequenceOfOperation);
```

Comment:
Do you need to check the return status from the GetControlSequenceOfOperation  ?

Reply:
I don't believe this is necessary.  The value for ControlSequenceOfOperation is initialized to a value outside of the valid range for the attribute.  If GetControlSequenceOfOperation() fails then the first if in the function will result in an error.